pyccel/codegen/printing/ccode.py

```diff
@@ -698,10 +698,14 @@ def _print_Allocate(self, expr):
         shape = ", ".join(a for a in shape)
         dtype = self._print(expr.variable.dtype)
         dtype = self.find_in_ndarray_type_registry(dtype, expr.variable.precision)
-        shape_Assign = "(int[]){" + shape + "}"
+        shape_dtype = self.find_in_dtype_registry('int', 4)
+        shape_Assign = "("+ shape_dtype +"[]){" + shape + "}"
         alloc_code = "{} = array_create({}, {}, {});".format(expr.variable, len(expr.shape), shape_Assign, dtype)
         return '{}\n{}'.format(free_code, alloc_code)
 
+    def _print_Deallocate(self, expr):
+        return 'free_array({})'.format(self._print(expr.variable.name))
```

Comment:
Why do you print `expr.variable.name` and not `expr.variable`? This looks like it is done to manage objects which are `stored_in_c_pointer`, but this shouldn't be the case for arrays should it?

Reply:
if the array have the attribute is_pointer == true he will be stored in `stored_in_c_pointer`

Reply:
and the views have the is_pointer set to true

Reply:
`stored_in_c_pointer` is a function which tells us whether the value is stored in a c pointer. So in this case if `stored_in_c_pointer` returns true then the type should be `t_ndarray*`. Is that right? If it isn't then it sounds like this needs to be corrected in `stored_in_c_pointer` not every time you use an array variable

Reply:
okey i think i will need to exclude the array from that function because I am not using the arrays as pointer at all so I can free them